File: src/Postman2CSharp.Wasm/wwwroot/json-editor/json-editor-interop.js
let jsonEditors = {};
function initJsonEditor(dotNetObjRef, id) {
    if (jsonEditors[id] != null) return;
    let jsonEditor = ace.edit(id);
    jsonEditor.setTheme("ace/theme/monokai");
    jsonEditor.session.setMode("ace/mode/json");
    jsonEditor.setShowPrintMargin(false);

    jsonEditor.on('change', function () {
        var json = jsonEditor.getValue();
        dotNetObjRef.invokeMethodAsync('CheckJsonValidity', json);
    });

    autoResize(jsonEditor);
    jsonEditors[id] = jsonEditor;
}

function getJsonEditorValue(id) {
    return jsonEditors[id] ? jsonEditors[id].getValue() : null;
}

function setJsonEditorValue(id, value) {
    if (jsonEditors[id]) {
        jsonEditors[id].setValue(value);
        jsonEditors[id].clearSelection();
    }
}

function resetJsonEditor(id) {
    if (jsonEditors[id]) {
        jsonEditors[id].setValue("");
    }
}

function destroyJsonEditor(id) {
    if (jsonEditors[id]) {
        jsonEditors[id].destroy();
        delete jsonEditors[id];
    }
}

function autoResize(editor) {
    if (!editor || !editor.container) {
        return;
    }
    var height = window.innerHeight;
    editor.container.style.height = `${(height * .6)}px`;
    editor.resize();
}

window.addEventListener('resize', function () {
    for (let id in jsonEditors) {
        autoResize(jsonEditors[id]);
    }
});

File: src/Postman2CSharp.UI/Services/JsonEditorInterop.cs
﻿using Microsoft.JSInterop;
using Postman2CSharp.UI.Shared;

namespace Postman2CSharp.UI.Services
{
    public class JsonEditorInterop
    {
        private readonly IJSRuntime _jsRuntime;

        public JsonEditorInterop(IJSRuntime jsRuntime)
        {
            _jsRuntime = jsRuntime;
        }

        public async Task Init(DotNetObjectReference<JsonEditor> dotNetObjRef, string id)
        {
            await _jsRuntime.InvokeVoidAsync("initJsonEditor", dotNetObjRef, id);
        }
        public async Task<string> GetValue(string id)
        {
            return await _jsRuntime.InvokeAsync<string>("getJsonEditorValue", id);
        }

        public async Task SetValue(string id, string value)
        { 
            await _jsRuntime.InvokeVoidAsync("setJsonEditorValue", id, value);
        }

        public async Task ResetValue(string id)
        {
            await _jsRuntime.InvokeVoidAsync("resetJsonEditor", id);
        }

        public async Task Destroy(string id)
        {
            await _jsRuntime.InvokeVoidAsync("destroyJsonEditor", id);
        }
    }
}
